tests/test_nms.cpp

```diff
@@ -57,3 +57,148 @@ TEST(TestSoftmax, Hello) {
   float output = softmax(input_vec);
   ASSERT_EQ(req_out, output);
 }
+
+TEST(TestGetNewOrder, Hello) {
+  std::vector<int> input_order{5, 4, 2, 1, 89};
+  std::vector<int> indices{0, 1, 2, 3};
+  std::vector<int> req_out{2, 1};
+
+  get_new_order(input_order, indices);
+  ASSERT_EQ(req_out, input_order);
+}
+
+bool floatCompare(float f1, float f2) {
+  static constexpr auto epsilon = 1.0e-01f;
+  if (fabs(f1 - f2) <= epsilon)
+    return true;
+  return false;
+}
+
+TEST(TestDistances, Hello) {
+  AnchorMat test_input = create_sample_input();
+
+  std::vector<int> x_indices = create_sequence_vector(
+      x_coordinate_start_idx, x_coordinate_start_idx + predication_steps - 1);
+  std::vector<int> z_indices = create_sequence_vector(
+      z_coordinate_start_idx, z_coordinate_start_idx + predication_steps - 1);
+
+  std::vector<std::vector<float>> all_x =
+      extract_columns<proposal_dim>(test_input, x_indices);
+  std::vector<std::vector<float>> all_z =
+      extract_columns<proposal_dim>(test_input, z_indices);
+
+  std::vector<float> expected_output = {0,        0.141421, 0.282843,
+                                        0.424264, 0.565685, 0.707107};
+  std::vector<float> accumulated_dis =
+      compute_all_lane_distance(all_x, all_z, all_x[0], all_z[0]);
+  for (int i = 0; i < accumulated_dis.size(); ++i) {
+    ASSERT_TRUE(floatCompare(accumulated_dis[i], expected_output[i]));
+  }
+  // ASSERT_EQ(req_out, input_order);
+}
+
+AnchorMat conf_testing() {
+  AnchorMat anchor_vec;
+
+  Anchor anchor_2{0.7, 1.3, 1.4, 0.6, 0.6};
+  Anchor anchor_3{0.8, 1.4, 1.3, 0.7, 0.5};
+  Anchor anchor_4{0.9, 1.5, 1.2, 0.8, 0.4};
+
+  anchor_vec.emplace_back(anchor_2);
+  anchor_vec.emplace_back(anchor_3);
+  anchor_vec.emplace_back(anchor_4);
+
+  return anchor_vec;
+}
+
+TEST(TestConfFiltering, Hello) {
+  AnchorMat test_input = create_sample_input();
+  AnchorMat expected_ouput = conf_testing();
+  auto [proposals, scores, anchor_inds] =
+      filter_proposals(test_input, 0.6, true);
+  for (int i = 0; i < expected_ouput.size(); ++i) {
+    for (int j = 0; j < expected_ouput[i].size(); ++j) {
+      ASSERT_TRUE(floatCompare(expected_ouput[i][j], proposals[i][j]));
+    }
+  }
+}
+
+TEST(TestNMS3D, Hello) {
+  AnchorMat test_input = create_sample_input();
+  auto [proposals, scores, anchor_inds] =
+      filter_proposals(test_input, 0.0, true);
+  NMS test_obj(test_input, 0.2, true);
+  std::vector<int> keep = test_obj.nms_3d(proposals, scores, 0.2);
+  std::vector<int> expected_out_indices{3, 1, 5};
+  ASSERT_EQ(keep.size(), expected_out_indices.size());
+  for (int i = 0; i < expected_out_indices.size(); ++i) {
+    ASSERT_EQ(expected_out_indices[i], keep[i]);
+  }
+}
+
+TEST(TestFilterRows, Hello) {
```

Comment:
What is the Hello supposed to represent in every test case ??

Reply:
Changes it